### ISSUE TITLE
Confusing error code/message when database is not loaded correctly

### ISSUE THREAD

**Issue by mkmik** [influxdata/influxdb_iox#1864]
__Steps to reproduce:__

1. Cause a database to not load correctly (e.g. by corrupting the catalog)

2. List the databases:

```console 
$ influxdb_iox database list | grep 844910ece80be8bc_64881f0f8fd1653c
844910ece80be8bc_64881f0f8fd1653c
```

3. Issue a query:

```console
$ influxdb_iox database query 844910ece80be8bc_64881f0f8fd1653c 'show tables'
Error querying: status: NotFound, message: "Database 844910ece80be8bc_64881f0f8fd1653c not found", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Thu, 01 Jul 2021 09:29:58 GMT"} }
```

__Expected behaviour:__

The `database list` should clearly show that the database is not healthy and/or the error code+message returned by the query api call should tell you that the database actually exists but it's broken.

__Details__

A call to the management API lets us see that this database exists but is not healthy:

```console
$ grpcurl --plaintext -d '{}' localhost:8082 influxdata.iox.management.v1.ManagementService.GetServerStatus | jq -r '.serverStatus.databaseStatuses[] | select(.dbName=="844910ece80be8bc_64881f0f8fd1653c")|.error.message'
cannot load catalog: Error during store write operation: AWS S3-based Object Store error: Unable to DELETE data. Bucket: toolsus1.a.influxcloud.net-iox, Location: 3003/844910ece80be8bc_64881f0f8fd1653c/transactions/00000000000000032490/9d0007d0-75df-41e2-b417-f72a0c6fae1d.txn, Error: Request ID: None Body: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>InternalError</Code><Message>We encountered an internal error. Please try again.</Message><RequestId>ME1H6HY1ZQG2SBFC</RequestId><HostId>61m56fSTRsgCSEsdydSqHheiQc9FJ4A2zt2ABVoGY0Izp2Cl12FsrpVOiX5BMoPxsUKCSIjaym4=</HostId></Error>
```

__Logs__

```
level=error msg="Error handling Flight gRPC request: Database 844910ece80be8bc_64881f0f8fd1653c not found" target="influxdb_iox::influxdb_ioxd::rpc::flight" location="src/influxdb_ioxd/rpc/flight.rs:73" time=1625132233439214991
```

CC: @crepererum

**Comment by mkmik** [comment]
Also, the db can be listed but its rules cannot be read:

```console
$ influxdb_iox database list | grep 844910ece80be8bc_64881f0f8fd1653c
844910ece80be8bc_64881f0f8fd1653c
$ influxdb_iox database get 844910ece80be8bc_64881f0f8fd1653c
Error getting database: Database not found
```

**Comment by mkmik** [comment]
In our staging environment a query node failed to initialize with the following error:

```
      {
        "dbName": "844910ece80be8bc_3c0bd4c89186ca89",
        "error": {
          "message": "Cannot replay: Cannot replay: Cannot store from sequencer 0 during replay: Hard buffer size limit reached"
        },
        "state": "DATABASE_STATE_REPLAY"
      },
```

We cannot inspect the configuration of the database with `database get`, because a database that failed to load always returns 404.

```
  Code: NotFound
  Message: Resource database/844910ece80be8bc_3c0bd4c89186ca89 not found
  Details:
```

**Comment by tustvold** [comment]
I'm currently looking into this class of funkiness

**Comment by tustvold** [comment]
This should be fixed now by the changes in #2203
